### ISSUE TITLE
File does not seem to be saved

### ISSUE THREAD

**Issue by bobweasel** [py-pdf/pypdf#1903]
After performing the write, I opened it in Adobe Acrobat Pro DC, and saw the edits are applied. However, when I try to close the file, I am promoted to save the file. If I click no, I would be prompted every time I open and close the file. However, when I try to attach the file in an email and preview it, the edits are no longer visible. This is solved by clicking yes on the save dialog instead, and I noticed that the file size increases after that.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Windows-10-10.0.19044-SP0


$ python -c "import pypdf;print(pypdf.__version__)"
3.10.0
```

## Code + PDF

This is a minimal, complete example that shows the issue:

```python
from pypdf import PdfReader, PdfWriter

reader = PdfReader("1.pdf")
writer = PdfWriter()

page = reader.pages[0]
fields = reader.get_fields()

writer.append(reader)

writer.update_page_form_field_values(
    writer.pages[0], {
        "Given Name Text Box": "asdf",
        "Family Name Text Box": "fam",
        "Address 1 Text Box": "add1",
        "House nr Text Box": "2",
        "Address 2 Text Box": "add2",
        "Postcode Text Box": "123456",
        "City Text Box": "city",
        "Country Combo Box": "Belgium",
        "Gender List Box": "Man",
        "Height Formatted Field": "30",
        "Driving License Check Box": "city",
        "Favourite Colour List Box": "Black",
        "Language 1 Check Box": "/Yes",
        "Language 2 Check Box": "/Off",
    }
)

# write "output" to pypdf-output.pdf
with open("filled-out.pdf", "wb") as output_stream:
    writer.write(output_stream)
```

Share here the PDF file(s) that cause the issue. The smaller they are, the
better. Let us know if we may add them to our tests!
http://foersom.com/net/HowTo/data/OoPdfFormExample.pdf

## Traceback

This is the complete Traceback I see:

No traceback

**Comment by pubpub-zz** [comment]
@bobweasel 
`.update_page_form_field_values()` has a new param `auto_regenerate` which affects need_appearances flag. it can take 3 values:
`True/False` which set/clear the flag and `None` to keep the old value
when `True` this flag as the viewer to refresh the appearance based on from the "/V" field and the document is asked to be saved.
Unfortunately for history reason reason the default value is `True`. add `,auto_regenerate = False` which should prevent the behavior you are observing.

**Comment by bobweasel** [comment]
It's not working, setting the value to `False` or `None` results in the values not showing up at all.

**Comment by kubaPod** [comment]
Just looking around for similar issue and tried your example.
It worked for me on Win 10 pypdf 3.12.0
Btw: `"Driving License Check Box": "city",` should be `"/Yes"`, right?

**Comment by pubpub-zz** [comment]
@kubaPod Can you publish your output pdf ?

**Comment by kubaPod** [comment]
@pubpub-zz sure:

[filled-out.pdf](https://github.com/py-pdf/pypdf/files/11964327/filled-out.pdf)

generated with:

```python
from pypdf import PdfReader, PdfWriter

reader = PdfReader("1.pdf")
writer = PdfWriter()

page = reader.pages[0]
fields = reader.get_fields()

writer.append(reader)

writer.update_page_form_field_values(
    writer.pages[0], {
        "Given Name Text Box": "asdf",
        "Family Name Text Box": "fam",
        "Address 1 Text Box": "add1",
        "House nr Text Box": "2",
        "Address 2 Text Box": "add2",
        "Postcode Text Box": "123456",
        "City Text Box": "city",
        "Country Combo Box": "Belgium",
        "Gender List Box": "Man",
        "Height Formatted Field": "30",
        "Driving License Check Box": "city",
        "Favourite Colour List Box": "Black",
        "Language 1 Check Box": "/Yes",
        "Language 2 Check Box": "/Off",
    },
    auto_regenerate = False
)

# write "output" to pypdf-output.pdf
with open("filled-out.pdf", "wb") as output_stream:
    writer.write(output_stream)
```

**Comment by pubpub-zz** [comment]
@bobweasel 
can you confirm you are getting the same results as @kubaPod so we can close this issue

**Comment by pubpub-zz** [comment]
@bobweasel :  +1?

**Comment by pubpub-zz** [comment]
I'm closing this issue until you can send the pdf else nothing can be done

**Comment by bobweasel** [comment]
@pubpub-zz just sent to the indicated email

**Comment by pubpub-zz** [comment]
@bobweasel 
Sorry for coming so late : I've found an strange thing in your file:  there is multiple spaces in the DA field of the annots after the font Inducing an issue that is now solved.

the PR is available if you want to test.

I've observed an odd result when filling the form  : the spacing is not respecting the 'cases'. when filling it with acrobat, I can observ some position shifts but I have not been able to understand how the viewer kwows what to do. 
I will open a new PR with a free form for people to help to find how to do it
